http-server-upload: add page (#7201)
tldr-pages/tldr: pages/common/http-server-upload.md

# http-server-upload

> Zero-configuration command-line HTTP server which provides a lightweight interface to upload files.
> More information: <https://github.com/crycode-de/http-server-upload>.

> - Start a HTTP server in the default port to upload files in the current directory:

Reviewer: Perhaps this will read better?
```suggestion
- Start a HTTP server on the default port to upload files to the current directory:
```

> - Start a HTTP server on the default port to upload files to the current directory:

Reviewer 2: It's `an HTTP`, not `a HTTP`.
Also, don't you mean `upload **from** the default directory`?

Author: No, the app uploads file vie an HTML form to the current directory.

Reviewer 2: I still don't get it. How can you upload something to your local machine? If you receive it you download, and if you send it you upload.

Author: You can start an HTTP server using this `npm` in your laptop or a cloud server, if you have access to that host&port from another PC you can enter to `192.168.0.10:8080` and then use this form with an `<input type="file" />` to upload files to another PC via a simple HTML form.
I used this mechanism a month ago to recover some files from a Windows 98 machine, I couldn't use a cloud storage because it's not compatible with really obsolete browsers, but this simple HTML form works 10/10.

`http-server-upload`

- Start a HTTP server with the specified maximum allowed file size for uploads in MiB (defaults to 200 MiB):

`MAX_FILE_SIZE={{size_in_megabytes}} http-server-upload`

> - Start a HTTP server in a specific port to upload files in the current directory:

Reviewer: Typo?
```suggestion
- Start a HTTP server on a specific port to upload files to the current directory:
```

> - Start a HTTP server on a specific port to upload files to the current directory:

Reviewer 2: Same here. `From the current directory`.

`PORT={{port}} http-server-upload`

> - Start a HTTP server storing the uploaded files into a specific directory:

Reviewer: Perhaps this will read better?
```suggestion
- Start a HTTP server, storing the uploaded files in a specific directory:
```

`UPLOAD_DIR={{path/to/directory}} http-server-upload`

> - Start a HTTP server using a specific directory to storing files during upload:

Reviewer: Perhaps this will read better?
```suggestion
- Start a HTTP server using a specific directory to temporarily store files during the upload process:
```

`UPLOAD_TMP_DIR={{path/to/directory}} http-server-upload`

> - Start a HTTP server accepting uploads with a specific token in the header:

Reviewer: A specific HTTP header, or any http header?

Author: Oh, the docs says token, in the posted form: 
```
curl \
  -F "uploads=@my-file.txt" \
  -F "uploads=@my-other-file.txt" \
  -F "path=my/dir" \
  -F "token=my-super-secret-token" \
  http://localhost:8080/upload
```

Reviewer: Hrm, in that case the example description here is wrong. Perhaps we could correct it then?

Author: ```suggestion
- Start a HTTP server accepting uploads with a specific token field in the HTTP post:
```
